mono/utils/mono-threads-windows.c

```diff
@@ -146,11 +137,20 @@ mono_threads_suspend_begin_async_resume (MonoThreadInfo *info)
 void
 mono_threads_suspend_register (MonoThreadInfo *info)
 {
+	BOOL success;
+	HANDLE currentThreadHandle = NULL;
+
+	success = DuplicateHandle(GetCurrentProcess(), GetCurrentThread(), GetCurrentProcess(), &currentThreadHandle, 0, FALSE, DUPLICATE_SAME_ACCESS);
+	g_assert(success && "Failed to duplicate current thread handle");
```

Comment:
`g_assertf(success, "Failed to duplicate current thread handle");`

Reply:
I'll correct this.

Reply:
I've fixed the assert at 06fa396.